pallets/weights/src/pallet_settlement.rs

```diff
@@ -0,0 +1,99 @@
+    }
+    fn add_instruction(l: u32) -> Weight {
+        (54_632_000 as Weight)
+            .saturating_add((48_594_000 as Weight).saturating_mul(l as Weight))
+            .saturating_add(DbWeight::get().reads(7 as Weight))
+            .saturating_add(DbWeight::get().reads((1 as Weight).saturating_mul(l as Weight)))
+            .saturating_add(DbWeight::get().writes(2 as Weight))
+            .saturating_add(DbWeight::get().writes((3 as Weight).saturating_mul(l as Weight)))
+    }
+    fn add_and_affirm_instruction(l: u32) -> Weight {
+        (302_559_000 as Weight)
+            .saturating_add((207_866_000 as Weight).saturating_mul(l as Weight))
+            .saturating_add(DbWeight::get().reads(4 as Weight))
+            .saturating_add(DbWeight::get().reads((5 as Weight).saturating_mul(l as Weight)))
+            .saturating_add(DbWeight::get().writes((6 as Weight).saturating_mul(l as Weight)))
+    }
+    fn set_venue_filtering() -> Weight {
+        (129_793_000 as Weight)
+            .saturating_add(DbWeight::get().reads(5 as Weight))
+            .saturating_add(DbWeight::get().writes(1 as Weight))
+    }
+    fn set_venue_filtering_disallow() -> Weight {
+        (123_320_000 as Weight)
+            .saturating_add(DbWeight::get().reads(5 as Weight))
+            .saturating_add(DbWeight::get().writes(1 as Weight))
+    }
+    fn allow_venues(v: u32) -> Weight {
+        (95_877_000 as Weight)
+            .saturating_add((6_462_000 as Weight).saturating_mul(v as Weight))
+            .saturating_add(DbWeight::get().reads(5 as Weight))
+            .saturating_add(DbWeight::get().writes((1 as Weight).saturating_mul(v as Weight)))
+    }
+    fn disallow_venues(v: u32) -> Weight {
+        (93_907_000 as Weight)
+            .saturating_add((6_054_000 as Weight).saturating_mul(v as Weight))
+            .saturating_add(DbWeight::get().reads(5 as Weight))
+            .saturating_add(DbWeight::get().writes((1 as Weight).saturating_mul(v as Weight)))
+    }
+    fn withdraw_affirmation(l: u32) -> Weight {
+        (5_414_000 as Weight)
+            .saturating_add((103_665_000 as Weight).saturating_mul(l as Weight))
+            .saturating_add(DbWeight::get().reads(4 as Weight))
+            .saturating_add(DbWeight::get().reads((5 as Weight).saturating_mul(l as Weight)))
+            .saturating_add(DbWeight::get().writes((4 as Weight).saturating_mul(l as Weight)))
+    }
+    fn withdraw_affirmation_with_receipt(l: u32) -> Weight {
+        (10_246_000 as Weight)
+            .saturating_add((106_451_000 as Weight).saturating_mul(l as Weight))
+            .saturating_add(DbWeight::get().reads(4 as Weight))
+            .saturating_add(DbWeight::get().reads((4 as Weight).saturating_mul(l as Weight)))
+            .saturating_add(DbWeight::get().writes((4 as Weight).saturating_mul(l as Weight)))
+    }
+    fn withdraw_affirmation_with_both_receipt_and_onchain_affirmation(l: u32) -> Weight {
+        (0 as Weight)
```

Comment:
My impression is that the base weight 0 is a consequence of the benchmark backend wrongly attributing the constant part of the weight to some of the complexity parameters. Since in this case there is only one parameter `l`, I think it would be fair to assume it skews the benchmark result. This is most likely because it ranges up to `u32::MAX` and we know that the backend selects the maximum in many and possibly most cases. It might be better to redo the benchmark with a fixed, large value for `l`.

(A cautionary remark about the use of complexity parameters: The distribution of values for complexity parameters in the backend is rather opinionated and definitely not uniform, which is bad for complexity analysis.)